gymnasium/envs/mujoco/f.py

```diff
@@ -0,0 +1,315 @@
+        reset_noise_scale: float = 0.1,
+        exclude_current_positions_from_observation: bool = True,
+        **kwargs,
+    ):
+        gymnasium.utils.EzPickle.__init__(
+            self,
+            xml_file,
+            frame_skip,
+            forward_reward_weight,
+            ctrl_cost_weight,
+            reset_noise_scale,
+            exclude_current_positions_from_observation,
+            **kwargs,
+        )
+
+        self._forward_reward_weight = forward_reward_weight
+        self._ctrl_cost_weight = ctrl_cost_weight
+
+        self._reset_noise_scale = reset_noise_scale
+
+        self._exclude_current_positions_from_observation = (
+            exclude_current_positions_from_observation
+        )
+
+        MJXEnv.__init__(
+            self,
+            model_path=xml_file,
+            frame_skip=frame_skip,
+            **kwargs,
+        )
+
+        obs_size = (
+            self.mjx_model.nq
+            + self.mjx_model.nv
+            - exclude_current_positions_from_observation
+        )
+
+        self.observation_space = gymnasium.spaces.Box(  # TODO use jnp when and if `Box` supports jax natively
+            low=-np.inf, high=np.inf, shape=(obs_size,), dtype=np.float32
+        )
+
+        self.observation_structure = {
+            "skipped_qpos": 1 * exclude_current_positions_from_observation,
+            "qpos": self.mjx_model.nq - 1 * exclude_current_positions_from_observation,
+            "qvel": self.mjx_model.nv,
+        }
+
+    def _gen_init_state(self, rng) -> tuple[jnp.ndarray, jnp.ndarray]:
+        noise_low = -self._reset_noise_scale
+        noise_high = self._reset_noise_scale
+
+        qpos = self.mjx_model.qpos0 + jax.random.uniform(
+            key=rng, minval=noise_low, maxval=noise_high, shape=(self.mjx_model.nq,)
+        )
+        qvel = self._reset_noise_scale * jax.random.normal(
+            key=rng, shape=(self.mjx_model.nv,)
+        )
+
+        return qpos, qvel
+
+    def observation(self, state: mjx._src.types.Data) -> jnp.ndarray:
+        mjx_data = state
+        position = mjx_data.qpos.flatten()
+        velocity = mjx_data.qvel.flatten()
+
+        if self._exclude_current_positions_from_observation:
+            position = position[1:]
+
+        observation = jnp.concatenate((position, velocity))
+        return observation
+
+    def _get_reward(
+        self,
+        state: mjx._src.types.Data,
+        action: jnp.ndarray,
+        next_state: mjx._src.types.Data,
+    ) -> tuple[jnp.ndarray, dict]:
+        mjx_data_old = state
+        mjx_data_new = next_state
+        x_position_before = mjx_data_old.qpos[0]
+        x_position_after = mjx_data_new.qpos[0]
+        x_velocity = (x_position_after - x_position_before) / self.dt
+
+        forward_reward = self._forward_reward_weight * x_velocity
+        ctrl_cost = self._ctrl_cost_weight * jnp.sum(jnp.square(action))
+
+        reward = forward_reward - ctrl_cost
+        reward_info = {
```

Comment:
This will be very useful for MO-Gymnasium :-)

Reply:
This was added in `Gymnasium/MuJoCo-v5` for all environments as per Lucas's requirements